### ISSUE TITLE
Incorrect check for iOS remote notifications permission 

### ISSUE THREAD

**Issue by ztepsa** [icerockdev/moko-permissions#64]
Library is using incorrect API to check iOS push notification permissions.

Following code uses `registeredForRemoteNotifications` to check current permission state, but this function in some cases always returns `true`, regardless if the app actually called `registerForRemoteNotifications`.
```kotlin
override fun isPermissionGranted(): Boolean {
    return UIApplication.sharedApplication().registeredForRemoteNotifications
}
```
[Apple docs](https://developer.apple.com/documentation/uikit/uiapplication/1623069-registeredforremotenotifications) say that `The value returned by this method takes into account the user’s preferences for receiving remote notifications` but I never seen iOS project use this call to check permission state. Also after a bit of testing in clean project this does not seem to be true.

Proper approach would be to always check authorization status from notifications settings. 
I added following function as temp solution (it just calls `getPermissionState`  and maps the output into `Boolean`)
```kotlin
suspend fun isPermissionGranted(permission: Permission): Boolean
```
Problem with this solution is that it does not work with current library API because `isPermissionGranted` is not a suspend function. I'm not really sure what would be a good solution to this problem, one idea that would work is to add semaphore (in case of remote notification check) and make a call to `getNotificationSettingsWithCompletionHandler` sync.

**Comment by petarmarijanovicfive** [comment]
+1! There are multiple issues with iOS Remote notifications, which makes this library for Kotlin Mobile development useless.

@Alex009 we've just merged the Android support for Remote notifications. Can we do the same for iOS as well 😁

**Comment by Alex009** [comment]
i think addition of `suspend` modifier better than addition of semaphore and blocking of thread.
can you send pull request with `getPermissionState` logic?

**Comment by Alex009** [comment]
will be released with 0.15.0
